homeassistant/components/yessssms/notify.py

```diff
@@ -8,34 +8,49 @@
 
 from homeassistant.components.notify import PLATFORM_SCHEMA, BaseNotificationService
 
+from .const import CONF_PROVIDER
+
 _LOGGER = logging.getLogger(__name__)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_USERNAME): cv.string,
         vol.Required(CONF_PASSWORD): cv.string,
         vol.Required(CONF_RECIPIENT): cv.string,
+        vol.Optional(CONF_PROVIDER, default="YESSS"): cv.string,
     }
 )
 
 
 def get_service(hass, config, discovery_info=None):
     """Get the YesssSMS notification service."""
     return YesssSMSNotificationService(
-        config[CONF_USERNAME], config[CONF_PASSWORD], config[CONF_RECIPIENT]
+        config[CONF_USERNAME],
+        config[CONF_PASSWORD],
+        config[CONF_RECIPIENT],
+        config[CONF_PROVIDER],
     )
 
 
 class YesssSMSNotificationService(BaseNotificationService):
     """Implement a notification service for the YesssSMS service."""
 
-    def __init__(self, username, password, recipient):
+    def __init__(self, username, password, recipient, provider=None):
         """Initialize the service."""
         from YesssSMS import YesssSMS
 
-        self.yesss = YesssSMS(username, password)
+        try:
+            self.yesss = YesssSMS(username, password, provider=provider)
```

Comment:
We don't want side effects in init method. Please move this to the `get_service` function and pass the client to the `YesssSMSNotificationService`.

Reply:
fixed it.